File: JavaScript/1018.js
// Problema: 1018 - Cédulas          | Resposta: Accepted
// Linguagem: JavaScript 12.18 [+2s] | Tempo: 0.028s

var input = require('fs').readFileSync('/dev/stdin', 'utf8');
const CEDULAS = [100, 50, 20, 10, 5, 2, 1];

function div(valor, divisor) {
    let qtde = parseInt(valor / divisor);
    let resto = valor % divisor;
    return [qtde, resto];
}

var dinheiro = resto = parseInt(input),
    qtde_cedulas;

console.log(dinheiro);
for (let nota of CEDULAS) {
    [qtde_cedulas, resto] = div(resto, nota);
    console.log(`${qtde_cedulas} nota(s) de R$ ${nota},00`);
}

File: Java/1018.java
// Problema: 1018 - Cédulas | Resposta: Accepted
// Linguagem: Java 19 [+2s] | Tempo: 0.097s

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	
	public static int[] resto(int valor, int nota) {
		int 
			n = (valor / nota),
			r = valor % nota;
		return new int[] { n, r };
	}
	
	static int[] cedulas = {100, 50, 20, 10, 5, 2, 1};

    public static void main(String[] args) throws IOException {
    	InputStreamReader ir = new InputStreamReader(System.in);
	    BufferedReader in = new BufferedReader(ir);
	    
	    int valor = Integer.parseInt(in.readLine());
	    System.out.println(valor);
	    for(int cedula : cedulas) {
	    	int[] resultado = resto(valor, cedula);
	    	valor = resultado[1];
	    	System.out.printf("%d nota(s) de R$ %d,00\n", resultado[0], cedula);
	    }
        
        in.close();
        ir.close();
    }
}